azurerm/resource_arm_storage_share.go

```diff
@@ -96,34 +105,70 @@ func resourceArmStorageShareRead(d *schema.ResourceData, meta interface{}) error
 		return nil
 	}
 
-	exists, err := resourceArmStorageShareExists(d, meta)
+	reference := fileClient.GetShareReference(name)
+	exists, err := reference.Exists()
 	if err != nil {
-		return err
+		return fmt.Errorf("Error testing existence of share %q: %s", name, err)
 	}
 
 	if !exists {
-		// Exists already removed this from state
-		return nil
+		log.Printf("[INFO] Share %q no longer exists, removing from state...", name)
+		d.SetId("")
 	}
 
-	name := d.Get("name").(string)
-
-	reference := fileClient.GetShareReference(name)
 	url := reference.URL()
 	if url == "" {
 		log.Printf("[INFO] URL for %q is empty", name)
 	}
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroupName)
+	d.Set("storage_account_name", storageAccountName)
 	d.Set("url", url)
 
+	reference.FetchAttributes(nil)
+	d.Set("quota", reference.Properties.Quota)
+
 	return nil
 }
 
+func resourceArmStorageShareUpdate(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+	ctx := armClient.StopContext
+
+	id := strings.Split(d.Id(), "/")
```

Comment:
there's a potential crash in these few lines if the segments aren't correct - as such we  probably want to add below this:

```
if len(id) != 3 {
  return fmt.Errorf("ID was not in the expected format - expected `{name}/{resourceGroup}/{storageAccountName}` got %q")
}
```

Reply:
Done